packages/platforms/mellanox/x86-64/x86-64-mlnx-msn2410/onlp/builds/src/module/src/fani.c

```diff
@@ -534,3 +534,14 @@ onlp_fani_ioctl(onlp_oid_t id, va_list vargs)
     return ONLP_STATUS_E_UNSUPPORTED;
 }
 
+int 
+onlp_fani_get_min_rpm(int id)
+{
+    int   len = 0, nbytes = 10;
+    char  r_data[10]   = {0};
+    char  fullpath[65] = {0};
+
+    snprintf(fullpath, sizeof(fullpath), "%s%s", PREFIX_PATH, fan_path[id].min);
+    OPEN_READ_FILE(fullpath, r_data, nbytes, len);
```

Comment:
Please concert all OPEN_READ_FILE macros to the onlp_file_read() API. 
It also supports format strings so you can avoid the snprintf/buffer.